users/error_handling.py

```diff
@@ -0,0 +1,50 @@
+import logging
+from functools import wraps
+
+from django.core.exceptions import ObjectDoesNotExist
+from django.db import IntegrityError
+from rest_framework import status
+from rest_framework.exceptions import AuthenticationFailed, NotFound, ParseError, PermissionDenied
+from rest_framework.response import Response
+
+logger = logging.getLogger(__name__)
+
+
+def handle_exceptions(func):
+    """
+    Unified decorator to handle specific exceptions uniformly across views, serializers, managers,
+    and general functions in Django.
+    """
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        try:
+
+            return func(*args, **kwargs)
+        except ObjectDoesNotExist as e:
+
+         return Response({'error': 'Resource not found', 'details': str(e)}, status=status.HTTP_404_NOT_FOUND)
```

Comment:
We can create ErrorMessages and SuccessMessages class and store all of these messages there